### ISSUE TITLE
move all.damaged handling to from repair mob code to generic code

### ISSUE THREAD

**Issue by rmsr** [sneezymud/sneezymud#109]
Currently all.damaged only works for the repair mobs, this useful shortcut ought to be usable anywhere.

**Comment by cizra** [comment]
Disagreed. Piggot is the expert on damaged things. I think it makes sense that he helps to identify things that can be improved.

**Comment by drewhershey** [comment]
From experience playing a warrior tank who had to go repair all my stuff
after every couple big fights, it would definitely have been nice to be
able to do a "remove all.damaged" to prepare for repairing my stuff back
up. Perhaps, following your logic, it could also be available to warriors
with a certain level of the blacksmithing skill since they would also be
experts on damaged things?

**Comment by rmsr** [comment]
Yes, player-repair skills is exactly the context I was thinking of.

**Comment by rmsr** [comment]
What if this was actually `all.repairable` and only worked for things the player knows how to repair?
